app/Models/Feed.php

```diff
@@ -316,26 +313,51 @@ public function load($loadDetails = false, $noCache = false) {
 
 				if (($mtime === true) || ($mtime > $this->lastUpdate) || $noCache) {
 					//Minz_Log::debug('FreshRSS no cache ' . $mtime . ' > ' . $this->lastUpdate . ' for ' . $clean_url);
-					$this->loadEntries($feed);	// et on charge les articles du flux
+					return $simplePie;
 				} else {
 					//Minz_Log::debug('FreshRSS use cache for ' . $clean_url);
-					$this->entries = array();
+					return null;
 				}
-
-				$feed->__destruct();	//http://simplepie.org/wiki/faq/i_m_getting_memory_leaks
-				unset($feed);
 			}
 		}
 	}
 
-	public function loadEntries($feed) {
-		$entries = array();
-		$guids = array();
+	public function loadGuids($simplePie) {
```

Comment:
We split the GUID logic from article logic, so that we can stream articles more cleanly